fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImpl.java

```diff
@@ -356,4 +396,104 @@ public Integer getTimeout() {
+     *     <li>decrements counter that keeps track of the outstanding (in flight) requests for the top level query,</li>
+     *     <li>and logs Druid response.</li>
+     * </ol>
+     *
+     * @param logCtx  The snapshot of the request log of the current thread
+     * @param timerName  The name that distinguishes this request as part of a druid query or segment metadata request
+     * @param outstanding  The counter that keeps track of the outstanding (in flight) requests for the top level query
+     * @param druidQueryId  The Druid query ID
+     * @param status  The response status
+     */
+    private void logRequest(
+            RequestLog logCtx,
+            String timerName,
+            AtomicLong outstanding,
+            String druidQueryId,
+            Status status
+    ) {
+        RequestLog.restore(logCtx);
+        RequestLog.stopTiming(timerName);
+        if (outstanding.decrementAndGet() == 0) {
+            RequestLog.startTiming(RESPONSE_WORKFLOW_TIMER);
+        }
+        RequestLog.record(new DruidResponse(druidQueryId));
+
+        LOG.debug(
+                "druid {} response code: {} {} and druid query id: {}",
+                getServiceConfig().getNameAndUrl(),
+                status.getStatusCode(),
+                status,
+                druidQueryId
+        );
+    }
+
+    /**
+     * Return true if response status code indicates an error.
+     *
+     * @param status  The Status object that contains status code to be checked
+     *
+     * @return true if the status code indicates an error
+     */
+    protected boolean hasError(Status status) {
+        return status != Status.OK;
+    }
+
+    /**
+     * Count and log error, then send error response.
+     *
+     * @param status  The response status
+     * @param response  The druid response
+     * @param druidQueryId  The Druid query ID
+     * @param error  callback for handling http errors.
+     */
+    private void markError(Status status, Response response, String druidQueryId, HttpErrorCallback error) {
+        getHttpErrorMeter().mark();
+        LOG.debug(
+                "druid {} error: {} {} {} and druid query id: {}",
+                getServiceConfig().getNameAndUrl(),
+                status.getStatusCode(),
+                status.getReasonPhrase(),
+                response.getResponseBody(),
+                druidQueryId
+        );
+
+        error.invoke(
+                status.getStatusCode(),
+                status.getReasonPhrase(),
+                response.getResponseBody()
+        );
+    }
+
+    /**
+     * Extract relevant information from response and wrap them into a JSON node.
+     *
+     * @param response  The response from which the information is to be retrieved.
+     *
+     * @return the relevant information in a JSON node
+     * @throws IOException when there is a JSON parsing error
+     */
+    protected JsonNode constructJsonResponse(Response response) throws IOException {
```

Comment:
This method is never used, and seems to have been replaced by the strategy pattern.